core/src/main/scala/chandu0101/scalajs/react/components/package.scala

```diff
@@ -1,14 +1,26 @@
 package chandu0101.scalajs.react
 
+import japgolly.scalajs.react.{BackendScope, Callback, CallbackTo, Ref}
 import scala.scalajs.js
 
 package object components
   extends util.CommonStyles
   with util.MTypes {
 
-  private[components] val JSMacro   = chandu0101.macros.tojs.JSMacro
+  private[components] val JSMacro = chandu0101.macros.tojs.JSMacro
 
   /* type alias for `js.undefined` */
-  private[components] type U[T] = js.UndefOr[T]
-  private[components] val uNone = js.undefined
+  type U[T] = js.UndefOr[T]
+  val uNone = js.undefined
+
+  def withRef[P, S](ref: Ref, t: => BackendScope[P, S])
```

Comment:
i wrote this little thing to handle references and callback handling, i was drowning in boilerplate